cinema/views.py

```diff
@@ -54,6 +89,35 @@ class MovieSessionViewSet(viewsets.ModelViewSet):
     queryset = MovieSession.objects.all()
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+
+        movie = self.request.query_params.get("movie")
+        date_string = self.request.query_params.get("date")
+
+        if movie:
+            movie_id = int(movie)
+            queryset = queryset.filter(movie_id=movie_id)
+
+        if date_string:
+            date = datetime.strptime(date_string, "%Y-%m-%d")
+            queryset = queryset.filter(
+                show_time__day=date.day,
+                show_time__year=date.year,
+                show_time__month=date.month
+            )
```

Comment:
`date` is already in the relevant format. You can just use `filter(show_time=date)`